File: src/go/internal/mouse/position.go
package mouse

import (
	"fmt"
	"runtime"
)

// Position represents a mouse position on the screen
type Position struct {
	X int
	Y int
}

// GetPosition returns the current mouse cursor position
func GetPosition() (Position, error) {
	if runtime.GOOS == "linux" {
		return getPositionLinux()
	}
	if runtime.GOOS == "windows" {
		return getPositionWindows()
	}
	return Position{}, fmt.Errorf("unsupported operating system: %s", runtime.GOOS)
}

// GetScreenSize returns the screen dimensions
func GetScreenSize() (width, height int, err error) {
	if runtime.GOOS == "linux" {
		return getScreenSizeLinux()
	}
	if runtime.GOOS == "windows" {
		return getScreenSizeWindows()
	}
	return 0, 0, fmt.Errorf("unsupported operating system: %s", runtime.GOOS)
}

File: src/typescript/cli/src/mouse/position.ts
// Mouse position tracking for Windows using robotjs

import robot from "robotjs";

/**
 * Mouse position coordinates
 */
export interface MousePosition {
    x: number;
    y: number;
}

/**
 * Screen dimensions
 */
export interface ScreenSize {
    width: number;
    height: number;
}

/**
 * Get the current mouse position
 */
export function getMousePosition(): MousePosition {
    const pos = robot.getMousePos();
    return {
        x: pos.x,
        y: pos.y,
    };
}

/**
 * Get the screen size
 */
export function getScreenSize(): ScreenSize {
    const size = robot.getScreenSize();
    return {
        width: size.width,
        height: size.height,
    };
}
